### PR TITLE
LibC: Separate arch dependent fenv functions

### PR DESCRIPTION
**Pull request by TomFinet**
Remove all functions with platform #if's from fenv, and add arch dependent implementations instead. 
The build system now selects the implementation based on the platform.

This PR addresses #22657.

### CODE REVIEW

File: Userland/Libraries/LibC/arch/x86_64/fenv.cpp [hunk 1]
```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright (c) 2021, Mițca Dumitru <dumitru0mitca@gmail.com>
```
Comment (ADKaster) [inline Userland/Libraries/LibC/arch/x86_64/fenv.cpp:2]: Could you update the year here to `2021-2024`

File: Userland/Libraries/LibC/arch/aarch64/fenv.cpp [hunk 2]
```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2021, Mițca Dumitru <dumitru0mitca@gmail.com>
```
Comment (ADKaster) [inline Userland/Libraries/LibC/arch/aarch64/fenv.cpp:2]: The aarch64 and riscv files seems different enough that you can put either your own copyright line, or, if you don't feel comfortable doing that (but you should feel comfortable :) ), the generic 'the SerenityOS Developers` line.

the year for these new files should just be 2024.

**Review by ADKaster — CHANGES_REQUESTED** [review]
Thanks for taking a crack at this issue! The new files seem much cleaner, thanks. I have a few comments about the copyright lines, and then we should be good to go on this.

File: Userland/Libraries/LibC/arch/aarch64/fenv.cpp [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+int feraiseexcept(int exceptions)
+{
+    fenv_t env;
+    fegetenv(&env);
+
+    exceptions &= FE_ALL_EXCEPT;
+
+    (void)exceptions;
+    TODO_RISCV64();
```
Comment (kleinesfilmroellchen) [inline Userland/Libraries/LibC/arch/aarch64/fenv.cpp:106]: TODO_AARCH64

**Review by kleinesfilmroellchen — CHANGES_REQUESTED** [review]
Thanks for doing this, even though I'm going to have a lot of fun rebasing the RISC-V implementation on top of this... just a small thing that slipped through.

**Review by ADKaster — APPROVED** [review]

**Pull request merged**